server/controller/processors/goalProcessor.ts

```diff
@@ -64,14 +70,21 @@ async function putGoalFeedback(req: Request, res: Response) {
         return;
     }
     res.sendStatus(StatusCode.OK);
+    const goal = await goalAPI.getGoalById(Number(req.params.id))
+    const goals = JSON.stringify(goal);
+    const module = await moduleAPI.getModuleById(Number(goals.replace("[{\"module_id\":", "").replace("}]", "")))
```

Comment:
I'm not entirely sure why you're using JSON.stringify() on the resulting goal array (which should always contain one Goal since every id should be unique) if all you need is the module_id. A much simpler way to achieve this is to call goal[0].module_id and add a check to see if module_id is undefined beforehand.